### PR TITLE
Style recommendations

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

Style recommendations to make code more Pythonic

## Does this introduce a breaking change?

I hope not :) I haven't tried to get it all running, however.

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[X] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

Probably following the README. I'm hoping it's easy for you to test.

### CODE REVIEW

File: api_blob_trigger/__init__.py [hunk 1]
```diff
@@ -10,24 +10,20 @@
...
+    logging.info("Python blob trigger function processed blob \nName: %s \nBlob Size: %s bytes", myblob.name, myblob.length)
 
-    logging.info("Start processing Bing News Search results for '%s'.", {myblob.name})
+    logging.info("Start processing Bing News Search results for '%s'.", myblob.name)
 
     # read the blob content as a string.
     search_results_blob_str = myblob.read()
 
-    # decode the string to Unicode, then replace single quotes with double quotes.
+    # decode the string to Unicode
```
Comment (pamelafox) [inline api_blob_trigger/__init__.py:20]: Updated comment to be accurate as we removed the replacement aspect

File: api_blob_trigger/__init__.py [hunk 2]
```diff
@@ -36,7 +32,7 @@ def main(myblob: func.InputStream):
         new_data_dictionary = clean_documents(data)
         logging.info("Successfully processed Bing News Search results for '%s'.", myblob.name)
 
-        new_json_str = str(json.dumps(new_data_dictionary))
+        new_json_str = json.dumps(new_data_dictionary)
```
Comment (pamelafox) [inline api_blob_trigger/__init__.py:35]: Calling str() shouldn't be needed

File: api_blob_trigger/__init__.py [hunk 3]
```diff
@@ -48,5 +44,5 @@ def main(myblob: func.InputStream):
             "Successfully uploaded to data lake, old: %s, new: %s", myblob.name, new_file_name
         )
 
-    except ValueError as e:
-        logging.info("Error converting %s to python dictionary: %s", myblob.name, e)
+    except ValueError as err:
```
Comment (pamelafox) [inline api_blob_trigger/__init__.py:47]: Pylint does not like single-letter names

File: api_search/__init__.py [hunk 4]
```diff
@@ -19,8 +19,8 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
 
     logging.info("/api/news")
 
-    search_term = req.params.get("search_term") or "Quantum Computing"
-    count = req.params.get("count") or 10
+    search_term = req.params.get("search_term", "Quantum Computing")
+    count = req.params.get("count", 10)
```
Comment (pamelafox) [inline api_search/__init__.py:23]: The second argument in get() can be used to specify a default fallback value

File: api_search/__init__.py [hunk 5]
```diff
@@ -41,14 +41,13 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
...
     if news_search_results.value:
-
         news_item_count = len(news_search_results.value)
-        logging.info("news item count: %s", str(news_item_count))
-
-        jsonItems = json.dumps([news.as_dict() for news in news_search_results.value])
+        logging.info("news item count: %d", news_item_count)
+        json_items = json.dumps([news.as_dict() for news in news_search_results.value])
 
+    # TODO: Why upload if there are no items?
```
Comment (pamelafox) [inline api_search/__init__.py:48]: Added this TOOD here- Given that json_items is inside an if condition, shouldnt the uploading also be inside the if?

File: shared/data_lake.py [hunk 6]
```diff
@@ -11,19 +11,16 @@ def upload_to_data_lake(azure_credential, file_name, data_str):
     container_name = os.environ.get("DATALAKE_GEN_2_CONTAINER_NAME")
     directory_name = os.environ.get("DATALAKE_GEN_2_DIRECTORY_NAME")
 
-    unique_file_path = f"{file_name}"
```
Comment (pamelafox) [inline shared/data_lake.py:14]: unique_file_path seems to be exact same as file_name so not sure why new variable is needed?

File: shared/hash.py [hunk 7]
```diff
@@ -3,5 +3,5 @@
 
 
 def get_random_hash():
-    hash = "".join(random.sample(string.ascii_letters + string.digits, 15))
-    return hash
+    random_hash = "".join(random.sample(string.ascii_letters + string.digits, 15))
```
Comment (pamelafox) [inline shared/hash.py:6]: hash is an existing name in Python, so it's best to not overwrite it.

File: shared/key_vault_secret.py [hunk 8]
```diff
@@ -5,6 +5,7 @@
 
 def get_key_vault_secret(azure_credential, key_vault_name, secret_name):
 
+    # TODO: Does this need to have hyphens stripped?
```
Comment (pamelafox) [inline shared/key_vault_secret.py:8]: I put this TODO here as I'm wondering if others will run into the hyphen problem that I ran into. I use hyphens all the time in my resource names.

**Comment by pamelafox** [comment]
@diberry Suggestions/comments for you! One overall observation is that there is *a lot* of logging. I wonder if all that logging is actually helpful, or if some can be removed. Up to you since logging can help someone debug something not working, but generally, in production code, we probably wouldn't want that much logging since it incurs storage and CPU costs. Also doubles the length of code. I typically only have logging if I'm debugging something and that's the only way to see what's happening.

File: api_search/__init__.py (hunk 5 above)
Comment (diberry) [reply api_search/__init__.py:48]: Good catch.

File: shared/data_lake.py (hunk 6 above)
Comment (diberry) [reply shared/data_lake.py:14]: Left over from an edit - good catch.

**Review by diberry — APPROVED** [review]

**Pull request merged**